services/core-api/app/api/utils/setup_marshmallow.py

```diff
@@ -53,65 +58,69 @@ class CoreConverter(ModelConverter):
 #         return super(SmartNested, self).serialize(attr, obj, accessor)
+AUDIT_COLUMNS = ('create_user', 'create_timestamp', 'update_user', 'update_timestamp')
+
+
+def run_after_configure():
+    setup_static_data(BaseModel)
+    setup_schema(BaseModel, db.session)()
 
 
 def setup_schema(Base, session):
     """
     inspired by: https://marshmallow-sqlalchemy.readthedocs.io/en/latest/recipes.html#automatically-generating-schemas-for-sqlalchemy-models
     """
     def setup_schema_fn():
-        for class_ in ActivityDetailBase.__subclasses__() + [Equipment, NOWApplicationDocumentXref]:
+        for class_ in ActivityDetailBase.__subclasses__() + [Equipment, NOWApplicationDocumentXref
+                                                             ] + sub_models.model_list:
             if hasattr(class_, "__tablename__"):
                 try:
                     if class_.__name__.endswith("Schema"):
                         raise ModelConversionError("For safety, setup_schema can not be used when a"
                                                    "Model class ends with 'Schema'")
+                    exclude_columns = AUDIT_COLUMNS if AuditMixin in class_.__bases__ else ()
 
                     class Meta(object):
                         model = class_
                         ordered = True
                         include_fk = True
                         sqla_session = db.session
                         model_converter = CoreConverter
-                        exclude = ('create_user', 'create_timestamp', 'update_user',
-                                   'update_timestamp')
-
+                        exclude = exclude_columns
+
+                    # After the schema is created on the class this looks for any schemas that have deferred the creation of
+                    # fields and validation. If found they are created and added to the proper schema.
+                    for k, v in class_._ModelSchema.__dict__.items():
+                        if type(v) == FieldTemplate:
+                            current_app.logger.debug(f'creating field for {k} on {class_}')
+                            class_._ModelSchema._declared_fields[k] = v.field(
+                                validate=validate.OneOf(choices=STATIC_DATA[v.one_of]))
                     schema_class_name = "%sSchema" % class_.__name__
-
-                    class ModelSchema2(ModelSchema):
-                        activity_type_code = fields.String(dump_only=True)
-                        activity_detail_id = fields.Integer(dump_only=True)
-
-                    schema_class = type(schema_class_name, (ModelSchema2, ), {"Meta": Meta})
+                    schema_class = type(schema_class_name, (class_._ModelSchema, ), {"Meta": Meta})
 
                     setattr(class_, "_schema", schema_class)
-                    current_app.logger.debug(f'created {schema_class}')
+                    current_app.logger.debug(f'created {class_}')
```

Comment:
can this be, created 'created schema for {class1}' or 'created {schema_class}'? just to be clear on what this does.

Reply:
Changed